### DISCUSSION TITLE
Help running snap on the server (SNAP_ORIGIN)

### DISCUSSION THREAD

**Discussion by devinvestidor** [MetaMask/snaps#1524] [General]
I'm using @metamask/template-snap-monorepo and I'm trying to connect snap on server, but my connection call: localhost. In README.md says to change .env.production to set SNAP_ORIGIN,

So I change SNAP_ORIGIN to DNS adress but in localhost I have this msg: MetaMask - RPC Error: Invalid snap ID: Expected the value to satisfy a union of intersection | string`, but received: "https://snapconnect.us/%22

And on server I have this msg: An error happened: Minified React error #321; visit https://reactjs.org/docs/error-decoder.html?invariant=321 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.

My .env.production: SNAP_ORIGIN=https://snapconnect.us/

Any sugest?

**Comment by Montoya** [comment]
The snap origin should be either localhost:8080 or an npm ID like `npm:pet-fox`. 

`https://snapconnect.us/` is your dapp, not your snap. 

Also if you have a repo I can look at, I can give you better advice.

**Reply by devinvestidor** [reply]
Thank you!!
I'm trying to use: local:http://localhost:8080 in ID, but it`s Ok in localhost, but not on the server. 

I thought I should change the SNAP_ID for the server.

**Repository:** https://github.com/blockful-io/snap-connect/tree/main/frontend

**Reply by Montoya** [reply]
Yes that's correct, on the server you cannot use localhost. But you need to publish your snap to npm and use `npm:[your npm ID]` for the server. 

Here's a tutorial: https://github.com/Montoya/random-snap#mystery-snap

**Reply by devinvestidor** [reply]
Thanks Brow! I will try add snap in npm. I'll let you know here soon.

**Comment by X-oss-byte** [comment]
Great thanks
